LightAPRS-hab/LightAPRS-hab.ino

```diff
@@ -106,90 +171,201 @@ void setup() {
+           gps.satellites.value() > 3) {
+
+          doAltCheck();
+          updateZone(); //updates BeaconWait
+          updateComment();
+          updatePosition();
+          updateTelemetry();
+          secsTillTx = BeaconWait;
+
+          //GpsOFF;
+          setGPS_PowerSaveMode();
+          //GpsFirstFix=true;
+          current_altitude = gps.altitude.feet();
+          if(current_altitude > max_altitude) {
+            max_altitude = current_altitude;
+          }
+          if(autoPathSizeHighAlt && gps.altitude.feet() > 3000){
             //force to use high altitude settings (WIDE2-n)
             APRS_setPathSize(1);
-        } else {
+          } else {
             //use defualt settings  
             APRS_setPathSize(pathSize);
-        }
-      
-      //send status message every 60 minutes
-      if(gps.time.minute() == 30){               
-        sendStatus();       
+          }
+          
+          //send status message every 15 minutes
+          if((gps.time.minute() % 15) == 0) {               
+            sendStatus();       
+          } else {
+            sendLocation();
+          }
+
+          freeMem();
+          Serial.flush();
+        } // if time to tx
+        // sleepSeconds(BeaconWait-((millis-loop_start)/1000));
       } else {
+#if defined(DEVMODE)
+      Serial.println(F("Not enough satelites"));
+#endif
+      }
+    }
 
-         sendLocation();
+    secsTillTx -= round((millis()-loop_start)/1000);
+  } else {
+    secsToCheckBatt--;
 
-      }
+    secsToCheckBatt -= (millis()-loop_start)/1000;
+    // sleepSeconds(BattWait-((millis-loop_start)/1000));
+  }
+  // Serial.println("Loop time in milliseconds->");
+  // Serial.println(round((millis()-loop_start)/1000));
+  sleepSeconds(secsTillTx);
+  secsTillTx = 0;
+}
 
-      freeMem();
-      Serial.flush();
-      sleepSeconds(BeaconWait);
 
-      } else {
-#if defined(DEVMODE)
-      Serial.println(F("Not enough sattelites"));
-#endif
-      }
-    } 
+
+// Begin smartPacket Functions
+void doAltCheck() {
+  if (gps.altitude.feet() <= lastalt) {
+    balloonDescendRepeat ++;
   } else {
+    balloonDescendRepeat = 0;
+  }
+  if (balloonDescendRepeat >= numDescendChecks) {
+    balloonPopped = true;
+  }
+}
+void updateZone() {
+  
+  long a = gps.altitude.feet();
+  
+  for (int i=0; i < NUM_ZONES; i++) {
+      if (zones[i].goingUp == true && balloonPopped == false) {
+        if (a <= zones[i].altitude) {
+          
+          currentSect = i;
+          BeaconWait = zones[i].secsForTx;
+          return;
 
-    sleepSeconds(BattWait);
-    
+        } else {
+          continue;
+        }
+      } else if (zones[i].goingUp == false && balloonPopped == true) {
+        
+          if (zones[i].altitude <= a) {
+            currentSect = i;
+            BeaconWait = zones[i].secsForTx;
+
+            return;
+          } else {
+          continue;
+          }
+      }
   }
   
+
 }
 
+void updateComment() {
+  comment[0] = ' ';
```

Comment:
in the next commit, let's change this by creating a string for the sprintf like this:
<set local variables for sprintf>
sprintf(comment, "U/D: %s XHU: %s%%  XTEMP: %s MX%03d", <variables>)

Reply:
Ok.